crates/zizmor/tests/integration/test-data/undocumented-permissions.yml

```
@@ -0,0 +1,44 @@
+name: Test Undocumented Permissions
+
+on: [push, pull_request]
+
+# This workflow-level permission block has no comment
+permissions:
+  contents: write
+  packages: read
+
+jobs:
+  test-job-1:
+    runs-on: ubuntu-latest
+    # This job's permissions block has a comment explaining why
+    permissions:
+      contents: read  # Need to read repo contents
+      issues: write   # Need to write issue comments
+    steps:
+      - uses: actions/checkout@v4
```

Nitpick: addressing the other findings here and below will make the snapshots a lot less noisy 🙂 

```suggestion
      - uses: actions/checkout@v4
        with:
          persist-credentials: false
```